src/modules/ekf2/ekf2_main.cpp

```diff
@@ -1109,29 +1059,33 @@ void Ekf2::run()
 					}
 				}
 
-				// Check and save the last valid calibration when we are disarmed
-				if ((vehicle_status.arming_state == vehicle_status_s::ARMING_STATE_STANDBY)
-				    && (status.filter_fault_flags == 0)
+				// Check and update the last valid magnetometer calibration
```

Comment:
It's not applied to the raw data (EKF behaviour is not changed in any manner). I just made the bias estimate available to other consumers without the need to land and disarm. It's just copied over to the `sensors_corrected` topic here : https://github.com/PX4/Firmware/pull/7523/files#diff-18e8f57d61a55777f8e25da991013704R965